task2/task2.Rmd

```diff
@@ -58,6 +58,26 @@ This task is considered to be finished when the chunk below is producing the out
 SELECT * FROM overview
 ```
 
+## hobos within 2km radius query
+
+```{sql connection=con}
+SELECT hobo_id,
+  st_astext(st_buffer(st_transform(geom, 25832), 2000)) as "hobos_within_2km"
```

Comment:
Good starting point! Be careful as your queries are only buffering the geometry of each hobo. Finally, you want to obtain the hobos **within** the buffers you just created.

Reply:
I tried to use something like this
`SELECT count(DISTINCT hobo_id FROM hobo WHERE st_within(st_transform(geom, 25832), st_buffer((SELECT st_transform(geom, 25832) FROM hobo), 2000))) as hobos_within_2km_2020
FROM hobos WHERE year = 2020`
But that only works when one tuple is selected e.g. WHERE id = 1
Is it possible to loop through all the values of the hobo_id col ?

Reply:
Looping in a sense like used in R is not really necessary. You can replace any value by an attribute name and as the query will be applied to the whole table row-wise anyway, the attribute name will be *replaced* by the value for that specific row.

I think what you want to do here, is to apply a filter condition to the nested query (the count part) that selects all **other** hobos within the buffer around the *current* geom attribute?

Maybe this helps. If you are still stuck, comment again, maybe with the current code, and I can further help you.